src/Elastic.Markdown/Myst/Directives/Diagram/DiagramRegistry.cs

```diff
@@ -0,0 +1,146 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.IO.Abstractions;
+
+namespace Elastic.Markdown.Myst.Directives.Diagram;
+
+/// <summary>
+/// Registry to track active diagrams and manage cleanup of outdated cached files
+/// </summary>
+public static class DiagramRegistry
+{
+	private static readonly HashSet<string> ActiveDiagrams = [];
+	private static readonly Lock Lock = new();
+
+	/// <summary>
+	/// Register a diagram as active during the current build
+	/// </summary>
+	/// <param name="localSvgPath">The local SVG path relative to output directory</param>
+	public static void RegisterDiagram(string localSvgPath)
+	{
+		if (string.IsNullOrEmpty(localSvgPath))
+			return;
+
+		lock (Lock)
+		{
+			_ = ActiveDiagrams.Add(localSvgPath);
+		}
+	}
+
+	/// <summary>
+	/// Get all currently registered active diagrams
+	/// </summary>
+	/// <returns>Collection of active diagram paths</returns>
+	public static IReadOnlyCollection<string> GetActiveDiagrams()
+	{
+		lock (Lock)
+		{
+			return ActiveDiagrams.ToArray();
+		}
+	}
+
+	/// <summary>
+	/// Clear all registered diagrams (typically called at start of build)
+	/// </summary>
+	public static void Clear()
+	{
+		lock (Lock)
+		{
+			ActiveDiagrams.Clear();
+		}
+	}
+
+	/// <summary>
+	/// Clean up unused diagram files from the output directory
+	/// </summary>
+	/// <param name="outputDirectory">The output directory path</param>
+	/// <returns>Number of files cleaned up</returns>
+	public static int CleanupUnusedDiagrams(string outputDirectory) =>
```

Comment:
Ensure the constructor takes `IFileSystem readFileSystem` and an `IFileSystem writeFileSystem` and use that for `CleanupUnusedDiagrams`